### PR TITLE
Make the Console handler a server component

### PR DESCRIPTION
**Pull request by RAOF**
The “ConsoleServices” concept is responsible for setting up a graphical terminal, notifying of switch events, and getting access to the hardware resources associated with the seat.

Currently the only non-trivial implementation is LinuxVirtualTerminal, but this paves the way for a logind implementation.

### CODE REVIEW

**Comment by RAOF** [comment]
bors try

File: src/platform/symbols.map [hunk 1]
```diff
@@ -160,3 +160,11 @@ MIRPLATFORM_1.0 {
     mir::options::wayland_socket_name_opt*;
   };
 } MIRPLATFORM_0.27;
+
+# Uuuuh, 1.0 is not yet a thing.
+MIR_PLATFORM_0.31 {
```
Comment (AlanGriffiths) [inline src/platform/symbols.map]: We've missed 0.31.

Although, as we're bumping ABI, can't we pull things together into MIR_PLATFORM_0.32?

Comment (RAOF) [reply src/platform/symbols.map]: Hm. Have we bumped mirplatform ABI? I don't think this changes any symbols from `libmirplatform`. It does bump the graphics platform ABI, but that is separate from the mirplatform ABI.

**Comment by RAOF** [comment]
The intent behind `obsolete_0_27` is primarily to make it easier for Ubiports; they don't need to update their existing platform modules in sync with the new Mir.

It was also to check that we *can* do this; both ABIs should be quite simple to support, so I did so to check that they would be simple to actually support.

The changes *are* a bit simpler if we drop the load-old-graphics-modules, so I can drop those if you like.

**Comment by RAOF** [comment]
@AlanGriffiths I've changed the title; as for the Mir-on-X11 crash - I think you've got an old `graphics-mesa-kms.so.13` hanging around in your libs/server-modules path that has the broken `probe()` which always detects that the platform-specific `--vt` option is set and so always returns `PlatformPriority::best`.

Hrm. Maybe I *should* just delete the load-obsolete-modules code

**Comment by RAOF** [comment]
Actually, looking at the code again, the Mir-on-X11 crash should *only* happen if you had built a previous version of this PR with `MIR_SERVER_GRAPHICS_PLATFORM_VERSION` bumped but not  `MIR_PLATFORM_GRAPHICS_ABI` bumped.

**Comment by AlanGriffiths** [comment]
> @RAOF Actually, looking at the code again, the Mir-on-X11 crash should only happen if you had built a previous version of this PR with MIR_SERVER_GRAPHICS_PLATFORM_VERSION bumped but not MIR_PLATFORM_GRAPHICS_ABI bumped.

I'll have a closer look at what is happening.

**Comment by RAOF** [comment]
Oh, right. The options are stringly typed, not namespaced at all.

This works for me, having copied a mesa-kms from trunk in to the built tree.

File: src/platform/graphics/platform_probe.cpp [hunk 2]
```diff
@@ -52,7 +52,15 @@ mir::graphics::module_for_device(
...
-                                return obsolete_probe(options);
+                                auto const priority = static_cast<unsigned int>(obsolete_probe(options));
+
+                                /*
+                                 * Cap obsolete modules to just less than PlatformPriority::supported.
+                                 * If *any* current module that will work, we want that instead.
+                                 */
+                                return priority >= PlatformPriority::supported ?
+                                    static_cast<PlatformPriority>(PlatformPriority::supported - 1) :
+                                    static_cast<PlatformPriority>(priority);
```
Comment (AlanGriffiths) [inline src/platform/graphics/platform_probe.cpp]: I think `std::min()` would be clearer. Vis:
```
                        return std::min(
                            obsolete_probe(options),
                            static_cast<PlatformPriority>(PlatformPriority::supported - 1));
```

**Pull request merged**